panel/widgets/input.py

```diff
@@ -370,6 +372,33 @@ def _process_param_change(self, msg):
         return msg
 
 
+class ArrayInput(LiteralInput):
+
+    max_array_size = param.Number(default=1000, doc="""
+        maximum number of elements in the array to allow full serialization
+        else only the string representation will be passed to the javascript
+        side and edition capabilities through the widget are lost.
```

Comment:
```suggestion
        Arrays larger than this limit will be allowed in Python but will not
        be serialized into JavaScript. Although such large arrays will thus
        not be editable in the widget, such a restriction helps avoid 
        overwhelming the browser and lets other widgets remain usable.
```

Also, what happens in the case of a large array? It doesn't look like the user will see any message, and I don't know what happens if the user does try to edit it in the browser. Ideally I'd think the widget should show the first few values along with a message indicating that it's truncated and not editable, and somehow any updating messages from the browser would be dropped if the user does edit it (and ideally they would not be allowed to edit it at all).